beep/tests/test_protocol.py

```diff
@@ -822,6 +822,40 @@ def test_schedule_creation(self):
             #                 os.path.join(scratch_dir, converted_sdu_name))
 
     def test_from_csv(self):
+        csv_file_list = os.path.join(TEST_FILE_DIR,
+                                     "data-share",
+                                     "raw",
+                                     "parameters",
+                                     "Iris_parameters - GP.csv")
+
+        with ScratchDir('.') as scratch_dir:
+            makedirs_p(os.path.join(scratch_dir, "procedures"))
+            makedirs_p(os.path.join(scratch_dir, "schedules"))
+            makedirs_p(os.path.join(scratch_dir, "names"))
+            generate_protocol_files_from_csv(csv_file_list, output_directory=scratch_dir)
+            if os.path.isfile(os.path.join(scratch_dir, "procedures", ".DS_Store")):
+                os.remove(os.path.join(scratch_dir, "procedures", ".DS_Store"))
+
+            procedure_list = os.listdir(os.path.join(scratch_dir, "procedures"))
+            for procedure_name in procedure_list:
+                sdu_test_input = os.path.join(SCHEDULE_TEMPLATE_DIR, "20170630-3_6C_9per_5C.sdu")
```

Comment:
If you encounter problems starting the file you might want to replace this `20170630-3_6C_9per_5C.sdu` file with a file that has already been used for the xIris cycling. There might be some additional parameters in the metadata that need to be set differently